source/ts/poll.ts

```diff
@@ -3,23 +3,35 @@ const DEFAULT_TIMEOUT = 10000;
 const wait = async (milliseconds: number) =>
   new Promise((resolve) => setTimeout(resolve, milliseconds));
 
+const queryTimeout = (timeoutMs: number) =>
```

Comment:
This method doesn't really know anything about queries. Maybe it could be `rejectAfter(timeout)`